### ISSUE TITLE
Feature request: CommonMarkSettings

### ISSUE THREAD

**Issue by jskeet** [hishamco/MarkdownViewEngine#3]
I need to be able to resolve some custom URIs ("noda-type://" etc) - I believe I can do this with [CommonMarkSettings](https://github.com/Knagis/CommonMark.NET/blob/master/CommonMark/CommonMarkSettings.cs) but there's no way of providing any in MarkdownViewEngine. This could be a property in `MarkdownViewEngineOptions`, then passed along in `MarkdownPage`.

**Comment by hishamco** [comment]
I used CommonMarkdown only for converting the markdown to html, regarding the feature that you requested I was thinking to support this via the ```MarkdownViewEngineOptions``` but this will affect all the views, if you a good plan for this you always welcome

**Comment by jskeet** [comment]
For my use case, that would be absolutely fine.

**Comment by hishamco** [comment]
Yep, but I'm looking for a general solution, not for particular project 😄

**Comment by jskeet** [comment]
I suspect app-wide `CommonMarkSettings` will be pretty common - a view-specific way of providing settings would be nice as well, perhaps as part of the directives.

**Comment by hishamco** [comment]
Assume we have the following:
```html
@page layout="_SiteLayout" title="Home"
```
how may the settings will be look like?!! coz we may have a lot of settings, if you have a good idea, please share it

**Comment by jskeet** [comment]
The Razor style would make sense to me:

```
@Page {
  Title = ...,
  Layout = ...,
  // Other settings here, and this is general purpose C#
}
```

**Comment by hishamco** [comment]
Regarding to me I want the directive as simple as possible, it's similar to what we have seen in ASP.NET WebForms
```html
<%@ Page attribute="value" [attribute="value"...] %>
```
So if the requested attribute is not related to the page behavior, I prefer to pass them here https://github.com/hishamco/MarkdownViewEngine/blob/master/samples/MarkdownViewEngineSample/Startup.cs#L13

**Comment by jskeet** [comment]
Yes, per-app settings should definitely be settable there - it's only if there are page-specific settings

**Comment by Knagis** [comment]
By the way, for app-wide settings you should be able to change the properties on the global `CommonMarkSettings.Default` and thus change the way it processes the markdown. 

It was made this way with the intention to allow application to override how a intermediate library uses CommonMark.NET.

**Comment by hishamco** [comment]
It is pleasure to hear from you @Knagis, the CommonMark.NET was helping me a lot.

The PR #7 https://github.com/hishamco/MarkdownViewEngine/blob/commonMarkdownSettings/src/MarkdownViewEngine/MarkdownViewEngineOptions.cs#L10 is achieve the requested feature, by I'm asking myself if we can expose an MDVE object that has the common properties in `CommonMarkSettings` that used by devs or keep it as it is.

Please if you have ideas regarding the whole MDVE I will be appreciate it
